org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaClientConnection.java

```diff
@@ -140,6 +150,38 @@ public boolean applyWorkspaceEdit(WorkspaceEdit edit){
 		return response.getApplied().booleanValue();
 	}
 
+	/**
+	 * @see {@link org.eclipse.lsp4j.services.LanguageClient#unregisterCapability(RegistrationParams)}
+	 */
+	public void unregisterCapability(UnregistrationParams params) {
+		client.unregisterCapability(params).join();
+	}
+
+	/**
+	 * @see {@link org.eclipse.lsp4j.services.LanguageClient#registerCapability(RegistrationParams)}
+	 */
+	public void registerCapability(RegistrationParams params) {
+		client.registerCapability(params).join();

```

Comment:
don't join, that makes the server unresponsive, waiting for a response that never comes. 

```
"pool-1-thread-1" #24 prio=5 os_prio=31 tid=0x00007f9926366800 nid=0xb303 waiting on condition [0x0000700009406000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000078120e1f0> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
	at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1934)
	at org.eclipse.jdt.ls.core.internal.JavaClientConnection.registerCapability(JavaClientConnection.java:156)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.didChangeConfiguration(JDTLanguageServer.java:197)

```

Not sure if this is a bug in LSP4J, i.e. maybe `registerCapability` should return a Future. cc @akosyakov @spoenemann